msteams-platform/messaging-extensions/build-api-based-message-extension.md

```diff
@@ -73,14 +113,206 @@ If the root object of the OpenAPI schema contains well-known array property name
+   { 
+   "id": "{Azure AD AppId}", 
+   "resource": "api://subdomain.example.com/botId-{guid}" 
+   }
+   ```
+
+1. **Add SSO Support to the Plugin**: Add the following auth section to the plugin part of the manifest. This indicates that the plugin supports SSO.
+
+   ```json
+   "authorization": { 
+     "authType": "microsoftEntra",
+     "microsoftEntraConfiguration": {
+         "supportsSingleSignOn": true,
+
+        }
+    },
+
+   ```
+
+1. **Validate the Token**: Before the token is sent to the plugin, validate that the resource URI and the domain the request is sent to are the same. Also, confirm the user ID in the token is the same as the one used for SMBA auth.
+
+1. **Send Invoke Request with Access Token**: The client sends an invoke request with the access token. An invoke request is a type of HTTP request that is used to trigger actions on the server. Here's an example payload that contains the access token.
+
+   ```json
+   {  
+   "name": "composeExtension/query",  
+   "value": {  
+      "commandId": "insertWiki",  
+      "parameters": [{  
+         "name": "searchKeyword",  
+         "value": "lakers"  
+      }],  
+      "authentication": {    
+         "token": "…",    
+      },    
+      "queryOptions": {  
+         "skip": 0,  
+         "count": 25  
+      }  
+   }
+   }
+   ```
+
+### Limitations
+
+This flow is valid only for consenting to a limited set of user-level APIs, such as email, profile, offline_access, and OpenId. It isn't used for other Graph scopes such as User.Read or Mail.Read. If you need to access other Graph scopes, you'll need to implement additional consent flows.
+
+```mermaid
+sequenceDiagram
+    participant TeamsApp as Teams App
+    participant AzureAD as Azure AD
+    participant AppServer as App Server
+    participant TeamsManifest as Teams Manifest
+    TeamsApp->>AzureAD: Create Azure AD app
+    AzureAD-->>TeamsApp: Return app ID and application ID URI
+    TeamsApp->>AppServer: Add code to handle access token
+    TeamsApp->>TeamsManifest: Update Teams app manifest with app ID and application ID URI
+    TeamsManifest-->>TeamsApp: Confirm update
+    Note over TeamsApp: Additional Changes
+    TeamsApp->>TeamsManifest: Add auth section to the manifest
+    TeamsManifest-->>TeamsApp: Confirm update
+    TeamsApp->>AppServer: Add field to auth config indicating SSO support
+    AppServer-->>TeamsApp: Confirm update
+    Note over TeamsApp: High-level Flow
+    TeamsApp->>AzureAD: Invoke Payload Containing the Token
+    AzureAD-->>TeamsApp: Return access token
+    TeamsApp->>AppServer: Send invoke request with access token
+    AppServer-->>TeamsApp: Validate token and confirm user id
+```
+
+This sequence diagram represents the flow of creating an Azure AD app, handling the access token, updating the Teams app manifest, and making additional changes to support SSO in Teams. It also includes the high-level flow of invoking a payload containing the token.
```

Comment:
It's not useful to include the setup steps like AAD app registration, manifest changes, etc. in this sequence diagram. Take that out please.
Please include only the interactions that happen when the API ME is invoked.

Reply:
removed the sequence diagram.